### PR TITLE
✅ Add validation rules for amp-autocomplete.

### PR DESCRIPTION
**Pull request by caroqliu**

### CODE REVIEW

File: extensions/amp-autocomplete/0.1/test/validator-amp-autocomplete.out [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+|          <script type="application/json"></script>
+|      </amp-autocomplete> 
+|
+|      <!-- Valid: amp-component with remote data and submit-on-enter attribute -->
+|      <amp-autocomplete submit-on-enter src="https://data.com/articles.json?ref=CANONICAL_URL">
+|        <input type="text">
+|      </amp-autocomplete> 
+|
+|      <!-- Valid: amp-component with static data and rich content template -->
+|      <amp-autocomplete>
```
Comment (cvializ) [inline extensions/amp-autocomplete/0.1/test/validator-amp-autocomplete.out]: It looks like some valid cases are coming up as invalid

**Comment by cvializ** [comment]
@honeybadgerdontcare (or @Gregable ): I discussed this with @caroqliu and we were both puzzling over how to accomplish the above validation logic. Any advice?

**Comment by honeybadgerdontcare** [comment]
@cvializ 	yes, this is something I'm looking into

**Comment by cvializ** [comment]
Thanks!

File: extensions/amp-autocomplete/validator-amp-autocomplete.protoascii [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  requires_extension: "amp-autocomplete"
+  mandatory_parent: "AMP-AUTOCOMPLETE"
+  attrs: { 
+    name: "nonce" 
+    disabled_by: "transformed"
+  }
+  attrs: {
+    name: "type"
+    mandatory: true
+    value: "application/json"
```
Comment (honeybadgerdontcare) [inline extensions/amp-autocomplete/validator-amp-autocomplete.protoascii]: We would want this to be `value_casei` instead of `value`.

**Comment by caroqliu** [comment]
Made the changes we discussed offline yesterday, ptal @cvializ

**Pull request merged**